src/components/feed/CreateFeed/ChrisFileSelect.tsx

```diff
@@ -0,0 +1,334 @@
+      // if it's a folder with matching children, show the folder
+      if (child.children) {
+        const folderShownChildren = this.computeVisibleChildren(child.children); // get its shown children
+        if (folderShownChildren.length) { // if it  has shown children
+          const folder = {
+            name: child.name,
+            path: child.path,
+            children: folderShownChildren,
+          }
+          shownChildren.push(folder);
+          continue; // do not re-evaluate folder once it's shown
+        }
+      }
+      if (this.matchesFilter(child.name)) { // is file or folder and matches
+        shownChildren.push(child);
+      }
+    }
+    return shownChildren;
+  }
+
+  resetVisibleTree() {
+    this.setState({ visibleTree: this.state.initialTree })
+  }
+
+  normalizeString(str: string) {
+    return str.toLowerCase().trim();
+  }
+
+  matchesFilter(name: string) {
+    return this.normalizeString(name).includes(this.normalizeString(this.state.filter));
+  }
+
+  // generates file name, with match highlighted, for file explorer
+  generateFileName(node: ChrisFile) {
+    const name = node.name;
+    if (!this.state.filter || !this.matchesFilter(name)) {
+      return name;
+    }
+    const matchIndex = this.normalizeString(name).indexOf(this.normalizeString(this.state.filter));
+    const before = name.substring(0, matchIndex);
+    const match = name.substring(matchIndex, matchIndex + this.state.filter.length);
+    const after = name.substring(matchIndex + this.state.filter.length);
+    return (
+      <React.Fragment>
+        {before}
+        <span className="match-highlight">{match}</span>
+        {after}
+      </React.Fragment>
+    );
+  }
+
+  renderTreeNode = (node: ChrisFile) => {
+    let isSelected;
+    const isFolder = !!node.children;
+    if (isFolder) {
+      // there can never be multiple folders with the same path, and folders don't have ids
+      isSelected = !!this.props.files.find(f => f.path === node.path);
+    } else {
+      // but there can be multiple files with the same path
+      isSelected = !!this.props.files.find(f => f.id === node.id);
+    }
+    const icon = isFolder
+      ? (node.collapsed ? <FolderCloseIcon /> : <FolderOpenIcon></FolderOpenIcon>)
+      : <FileIcon />
+    return (
+      <span className="name">
+        <Checkbox
+          checked={isSelected}
+          id={`check-${node.path}`}
+          aria-label=""
+          onChange={isChecked => this.handleCheckboxChange(isChecked, node)}
+        />
+        {icon}
+        {this.generateFileName(node)}
+      </span>
+    )
+  }
+
+  render() {
+    
+    const fileList = this.props.files.map(file => (
+      <div className="file-preview" key={file.path}>
+        {
+          file.children ? <FolderCloseIcon /> : <FileIcon />
+        }
+        <span className="file-name">{file.name}</span>
+        <CloseIcon className="file-remove" onClick={() => this.props.handleFileRemove(file) } />
```

Comment:
If you can bind this in the constructor and call this on click function without creating an anonymous function it would be more performant.

Reply:
Since `handleFileRemove` takes the `file` object as a parameter, I think it would need to be bound within the loop like `onClick={this.props.handleFileRemove.bind(this, file)`. Is that ok/better than what it currently is?

Reply:
@matankb the way it is looks ok. thanks!